geps/gep-3539/index.md

```diff
@@ -0,0 +1,200 @@
+routing in Kubernetes. Very simply, it describes how to get traffic into pods. 
+ClusterIP provides similar functionality of an ingress point for routing traffic 
+into pods. As the Gateway API has evolved, there have been discussions around whether 
+it can be a substitute for the increasingly complex and overloaded Service API. This 
+document aims to describe what this could look like in practice, with a focus on 
+ClusterIP and a brief commentary on how the concept design can be extended to 
+accommodate LoadBalancer and NodePort Services.
+
+## Overview
+
+Gateway API can be thought of as decomposing Service API into multiple separable 
+components that allow for definition of the ClusterIP address and listener configuration 
+(Gateway resource), implementation specifics and common configuration (GatewayClass 
+resource), and routing traffic to backends (Route resource).
+
+### Limitations of Service API
+
+Besides what has been discussed in the past about Service API maintainability, evolvability, 
+and complexity concerns, see: https://www.youtube.com/watch?v=Oslwx3hj2Eg, we ran into 
+additional practical concerns that rendered Service API insufficient for the needs at hand.
+
+Service IPs can only be assigned out of the ServiceCIDR range configured for the API server. 
+While Kubernetes 1.31 added a Beta feature that allows for the Extension of Service IP Ranges, 
+there have been use cases where multi-NIC pods (pods with multiple network interfaces) require 
+the flexibility of specifying different ServiceCIDR ranges to be used for ClusterIP services 
+corresponding to the multiple different networks. There are strict traffic splitting and network 
+isolation requirements that demand non-overlapping ServiceCIDR ranges for per-network ClusterIP 
+service groups. Because of the way service definition and IP address allocation are tightly 
+coupled in API server, it is not possible to use the current Service API to achieve this model 
+without resorting to inelegant and klugey implementations.
+
+Gateway API also satisfies, in a user-friendly and uncomplicated manner, the need for advanced 
+routing and load balancing capabilities in order to enable canary rollouts, weighted traffic 
+distribution, isolation of access and configuration.
+
+### Service Model to Gateway API Model
+
+![image displaying service model to gateway api model mapping](images/service-model-gw-api-model.png "image displaying service model to gateway api model mapping")
+
+### EndpointSelector as Backend
+
+A Route can forward traffic to the endpoints selected via selector rules defined in EndpointSelector. 
+While Service is the default resource kind of the referent in backendRef, EndpointSelector is 
+suggested as an example of a custom resource that implementations could have to attach pods (or 
+potentially other resource kinds) directly to a Route via backendRef. 
+
+```yaml
+{% include 'standard/clusterip-gateway/tcproute-with-endpointselector.yaml' %}
```

Comment:
these does not render. or does it suppose to be like this? see https://github.com/ptrivedi/gateway-api/blob/gep-clusterip-gateway/geps/gep-3539/index.md

Reply:
Oh, wasnt aware of that page.